### PR TITLE
print-routes options and print-plugins

### PR DESCRIPTION
**Pull request by bcomnes**
This adds a `method`, `common-prefix` and `include-hooks` flags to the `print-routes` command, and also implements a `print-plugins` flag, with no flags. 

- Closes https://github.com/fastify/fastify-cli/issues/636
- Closes https://github.com/fastify/fastify-cli/issues/635

Feedback wanted!

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: test/print-routes.test.js [hunk 1]
```diff
@@ -19,20 +20,16 @@ test('should print routes', async t => {
 
   await fastify.close()
   t.ok(spy.called)
-  t.same(spy.args, [['debug', '└── / (GET)\n    / (HEAD)\n    / (POST)\n']])
+  t.same(spy.args, [['debug', '└── / (GET, HEAD, POST)\n']])
 })
 
 test('should print routes via cli', async t => {
```
Comment (bcomnes) [inline test/print-routes.test.js:26]: This test seemed to do the exact same thing as the prior test, so I re-wrote it to actually run through the CLI. Hope this is right.

**Comment by bcomnes** [comment]
@mcollina Looks like tests aren't passing on windows because of the inline ENV var setting on the typescript esm unit test:

https://github.com/fastify/fastify-cli/blob/f111b25b96d9f29d42c9a8f5ced4108f56a073d0/package.json#L12

Want me to add cross-env to devDeps and make that work in the windows CI as well?

File: test/generate-esm.test.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ javascriptTemplate.type = 'module'
 javascriptTemplate.tap = {
   coverage: false
 }
-javascriptTemplate.devDependencies['c8'] = cliPkg.devDependencies['c8']
```
Comment (bcomnes) [inline test/generate-esm.test.js:36]: Lint:fix did this

**Comment by bcomnes** [comment]
There seems to be some kind of spurious timeout in some of these tests.

**Comment by mcollina** [comment]
It seems the failures are relevant: 

```
test/generate.test.js 1> run 'npm test' to execute the unit tests
​ FAIL ​ test/print-plugins.test.js
 ✖ timeout!

  test: test/print-plugins.test.js should plugins routes via cli
  expired: test/print-plugins.test.js

​ FAIL ​ test/print-plugins.test.js
 ✖ timeout!
```

**Comment by bcomnes** [comment]
Oh shoot I missed those. Looking

**Comment by bcomnes** [comment]
Ok @mcollina got this green. A couple of those tests are still flakey. I tried to re-order some event ordering tests but it still doesn't seem to fix it.

**Comment by big-kahuna-burger** [comment]
Not flakey anymore with #681

**Comment by bcomnes** [comment]
> @bcomnes 
> 
> Check #681
> 
> 

Great! Thank you

**Pull request merged**